node/src/store/indexed_db_store.rs

```diff
@@ -0,0 +1,533 @@
+use async_trait::async_trait;
+use celestia_types::hash::Hash;
+use celestia_types::ExtendedHeader;
+use rexie::{Direction, Index, KeyRange, ObjectStore, Rexie, TransactionMode};
+use send_wrapper::SendWrapper;
+use serde::{Deserialize, Serialize};
+use serde_wasm_bindgen::{from_value, to_value};
+
+use crate::store::{Result, Store, StoreError};
+
+const DB_VERSION: u32 = 1;
+const HEADER_STORE_NAME: &str = "headers";
+const HASH_INDEX_NAME: &str = "hash";
+const HEIGHT_INDEX_NAME: &str = "height";
+
+#[derive(Debug, Serialize, Deserialize)]
+struct ExtendedHeaderEntry {
+    height: u64,
+    hash: Hash,
+    header: ExtendedHeader,
+}
+
+// SendWrapper usage is safe in wasm because we're running on single thread
+#[derive(Debug)]
+pub struct IndexedDbStore(SendWrapper<Rexie>);
+
+impl IndexedDbStore {
+    pub async fn new_with_name(name: &str) -> rexie::Result<Self> {
+        let rexie = Rexie::builder(name)
+            .version(DB_VERSION)
+            .add_object_store(
+                ObjectStore::new(HEADER_STORE_NAME)
+                    .key_path("id")
+                    .auto_increment(true)
+                    .add_index(Index::new(HASH_INDEX_NAME, "hash").unique(true))
+                    .add_index(Index::new(HEIGHT_INDEX_NAME, "height").unique(true)),
+            )
+            .build()
+            .await?;
+        Ok(Self(SendWrapper::new(rexie)))
+    }
+
+    pub async fn new(network_id: &str) -> rexie::Result<Self> {
+        Self::new_with_name(&network_id).await
+    }
```

Comment:
I think this needs to be removed and rename `new_with_name` to `new`

Reply:
Removed, left just single `fn new(network_id: &str)` constructor.